File: MyNativeModule/ios/MyNativeModule/MyView.m
//
//  MyView.m
//  MyNativeModule
//
//  Created by Phong Nguyễn Hoàng on 26/03/2024.
//
#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(MyViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(backGroundColor, NSInteger)
  RCT_EXPORT_VIEW_PROPERTY(mySrc, NSString)
  RCT_EXPORT_VIEW_PROPERTY(myArr, NSArray)
@end

File: MyNativeModule/ios/MyNativeModule/MyView.swift
//
//  MyView.swift
//  MyNativeModule
//
//  Created by Phong Nguyễn Hoàng on 26/03/2024.
//

import UIKit

class MyView: UIImageView {
  override func didMoveToWindow() {
    contentMode = .scaleAspectFit
  }
  
  @objc var backGroundColor: Int = 0 {
    didSet {
      if backGroundColor == 0 {
        backgroundColor = .blue
      } else {
        backgroundColor = .green
      }
    }
  }
  
  @objc var mySrc: String = "" {
    didSet {
      if mySrc == "medie" {
        image = UIImage(named: "medie")
      } else {
        image = UIImage(named: "soul-time")
      }
    }
  }
  
  @objc var myArr: NSArray = [] {
    didSet{
      print(myArr)
    }
  }
}
